modelmapper/stats.py

```diff
@@ -323,14 +324,29 @@ class DateTimeMatcher(TypeMatcher, TypeAccumulator):
     """
     value_type = HasDateTime
 
-    def __init__(self, datetime_formats=None):
+    def __init__(self, datetime_formats=None, datetime_allowed_characters=None):
         self.datetime_formats = datetime_formats
+        self.datetime_allowed_characters = datetime_allowed_characters
         self.reset()
 
     def reset(self):
         self.candidate_formats = self.datetime_formats.copy()
         self.has_matched_before = False
 
+    def _may_need_new_datetime_format(self, item):
```

Comment:
Small nit. 

`may` to me would mean a type of `Optional[bool]`. Perhaps just name this `needs_new_datetime_format` since `true` and `false` are both valid answers.

Reply:
ok